tests/test_server.py

```diff
@@ -1,46 +1,65 @@
-import json
-
-import connexion
 import pytest
+import requests
 
-server = connexion.FlaskApp(__name__,
-                            specification_dir='../src/ratbag_emu/openapi/',
-                            debug=True)
-server.add_api('ratbag-emu.yaml',
-                options={"swagger_ui": True},
-                arguments={'title': 'ratbag-emu'},
-                strict_validation=True,
-                validate_responses=True)
 
+class Client(object):
+    def __init__(self, url='http://localhost:8080'):
+        self.url = url
 
-def test_permissions():
-    from hidtools.uhid import UHIDDevice
+    def get(self, path):
+        return requests.get(f'{self.url}{path}')
 
-    try:
-        assert UHIDDevice() is not None
-    except PermissionError:
-        pytest.exit('Not enough permissions to create UHID devices')
+    def post(self, path, data=None, json=None):
+        return requests.post(f'{self.url}{path}',
+                             data=data,
+                             json=json)
 
+    def delete(self, path):
+        return requests.delete(f'{self.url}{path}')
 
-@pytest.fixture(autouse=True)
-def handle_devices():
-    import threading
+    def put(self, path, data=None, json=None):
+        return requests.put(f'{self.url}{path}',
+                            data=data,
+                            json=json)
 
-    from . import ratbag_emu
 
-    import ratbag_emu.server
-    from ratbag_emu.device_handler import DeviceHandler
+def reload_udev_rules():
+    import subprocess
+
+    subprocess.run('udevadm control --reload-rules'.split())
+    subprocess.run('udevadm hwdb --update'.split())
+
+
+@pytest.fixture(scope='session', autouse=True)
+def udev_rules():
+    import os
+
+    from shutil import copyfile
+    from pathlib import Path
 
-    devices_thread = threading.Thread(target=DeviceHandler.handle)
-    devices_thread.daemon = True
-    devices_thread.start()
+    os.makedirs('/run/udev/rules.d', exist_ok=True)
+    copyfile('rules.d/61-ratbag-emu-ignore-test-devices.rules', '/run/udev/rules.d/61-ratbag-emu-ignore-test-devices.rules')
+    reload_udev_rules()
 
-    yield devices_thread
+    yield
+
+    rules = Path('/run/udev/rules.d/61-ratbag-emu-test-devices.rules')
+    if rules.is_file():
+        rules.unlink()
+        reload_udev_rules()
 
 
 @pytest.fixture(autouse=True)
-def client(handle_devices):
-    yield server.app.test_client()
+def client():
+    yield Client()
+
+
+def add_device(client, name='steelseries-rival310'):
+    data = {
+        'shortname': name
+    }
+    response = client.post('/devices/add', json=data)
+    return response.json()['id']
```

Comment:
Aren't you supposed to check if the answer is `200`?

Reply:
`201`, but yes. I forgot.